File: app/src/main/java/net/north101/android/ghplayertracker/data/CardSpecial.java
package net.north101.android.ghplayertracker.data;

import org.json.JSONException;

public enum CardSpecial {
    Shuffle,
    Rolling,
    Remove;

    public static CardSpecial parse(String data) throws JSONException {
        if (data == null) return null;

        switch (data) {
            case "shuffle":
                return CardSpecial.Shuffle;

            case "rolling":
                return CardSpecial.Rolling;

            case "remove":
                return CardSpecial.Remove;

            default:
                throw new JSONException("");
        }
    }
}

File: app/src/main/java/net/north101/android/ghplayertracker/data/CardSpecial.kt
package net.north101.android.ghplayertracker.data

import org.json.JSONException

enum class CardSpecial {
    None,
    Shuffle,
    Rolling,
    Remove;

    companion object {
        @Throws(JSONException::class)
        fun parse(data: String?): CardSpecial {
            if (data == null) return None

            return when (data) {
                "shuffle" -> CardSpecial.Shuffle
                "rolling" -> CardSpecial.Rolling
                "remove" -> CardSpecial.Remove
                else -> throw JSONException(data)
            }
        }
    }
}
